Server/server/models/WaitingRoom.ts

```diff
@@ -0,0 +1,20 @@
+import { Player } from './Player';
+
+export class WaitingRoom {
+
+    private roomID: number;
+    private players: Player[];
+
+    constructor(roomID: number) {
+        this.roomID = roomID;
+        this.players = [];
+    }
+
+    public addPlayer = (player: Player) => {
```

Comment:
The only place where I don't like using arrow functions is inside class definitions, because when this code is transpiled by tsc/babel, it does some funky stuff and things like inheritance are affected as a result. So just change each method definition to something like
```ts
public addPlayer(player: Player): void {
```